modules/google-maps/src/utils.js

```diff
@@ -0,0 +1,140 @@
+/* global document, google */
+import {Deck} from '@deck.gl/core';
+
+export function createDeckInstance(map, overlay, deck) {
+  if (deck) {
+    if (deck.props.userData.map === map) {
+      return deck;
+    }
+    // deck instance was created for a different map
+    destroyDeckInstance(deck);
+  }
+
+  const eventListeners = {
+    click: null,
+    mousemove: null,
+    mouseout: null
+  };
+
+  deck = new Deck({
+    canvas: createDeckCanvas(overlay),
+    initialViewState: {
+      longitude: 0,
+      latitude: 0,
+      zoom: 1
+    },
+    controller: false,
+    userData: {
+      map,
+      eventListeners
+    }
+  });
+
+  // Register event listeners
+  for (const eventType in eventListeners) {
+    eventListeners[eventType] = map.addListener(eventType, evt =>
+      handleMouseEvent(deck, eventType, evt)
+    );
+  }
+
+  return deck;
+}
+
+function createDeckCanvas(overlay) {
```

Comment:
Is that a `googleMapsOverlay`? Naming the parameter or adding comments about what the parameters are here (e.g. what type) would have helped my read-through.